terratools/terra_model.py

```diff
@@ -1572,6 +1587,42 @@ def add_geog_flow(self):
         self.set_field(field="u_enu", values=flow_enu)
 
 
+class TerraModelLayer(TerraModel):
+    """
+    A subclass of the TerraModel superclass, TerraModelLayer is for storing 2D layer
+    information which is written out of a TERRA simulation. Typically this might be some
+    boundary information, eg CMB heat flux or radial surface radial stresses, but could be
+    from any radial layer of the simulation in principle.
+
+    Methods of the TerraModel class which are not compatible with TerraModelLayer are
+    overwritten and will raise a LayerMethodError exception
+    """
+
+    def add_adiabat(self):
+        raise LayerMethodError(self.add_adiabat.__name__)
+
+    def get_1d_profile(self, field, lat, lon):
+        raise LayerMethodError(self.get_1d_profile.__name__)
+
+    def plot_section(
```

Comment:
Maybe just replace this with `plot_section(*args, **kwargs)`, since we want to raise a
`LayerMethodError` regardless of how it's called I guess?

Reply:
Yeah agreed, I'll change the above and add the docs into the PR